### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by AjroudiRida**
In this featured branch, I did the following:
-Add column species to animals Table
- Insert data
- Update the animals table 
- Commit the transaction.

code

- Delete data

code1

- Create a savepoint for the transaction.

code2

-Write queries using SQL aggregate functions

code3

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,5 +7,6 @@ CREATE TABLE animals (
     escape_attempts integer,
     neutered boolean,
     weight_kg decimal,
+    species text,
```
Comment (mckent05) [inline schema.sql]: - You have done well so far, but when altering your schema, it is advised to not directly edit your schema, please use SQL commands like:
   ```ALTER TABLE animals ADD species varchar(255);```. Please fix this

File: data.sql [hunk 2]
```diff
@@ -4,4 +4,11 @@ INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) V
...
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (1,'Charmander',TO_DATE('Feb 8, 2020', 'Mon DD, YYYY'),0,false,-11);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (2,'Plantmon',TO_DATE('Nov 15, 2021', 'Mon DD, YYYY'),2,true,-5.7);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (3,'Squirtle',TO_DATE('Apr 2, 1993', 'Mon DD, YYYY'),3,false,-12.13);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (4,'Angemon',TO_DATE('Jun 12, 2005', 'Mon DD, YYYY'),1,true,-45);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (5,'Boarmon',TO_DATE('Jun 7, 2005', 'Mon DD, YYYY'),7,true,20.4);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (6,'Blossom',TO_DATE('Oct 13, 1998', 'Mon DD, YYYY'),3,true,17);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES (7,'Ditto',TO_DATE('May 14, 2022', 'Mon DD, YYYY'),4,true,22);
```
Comment (mckent05) [inline data.sql]: - You have done well so far, but by manually inputting your animal ID, you have overwritten the previous data you had in your database, please correct this as it is crucial that you have the data from the earlier milestone of your database.
**OPTIONAL** - Please consider making your animal ID increment automatically so you do not have to input the ID for every animal manually 😄

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,42 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+UPDATE animals set weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+SELECT * FROM animals;
+COMMIT;
+
+SELECT COUNT(*) as nb_animals FROM animals;
+SELECT COUNT(*) as nb_animals FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) as average_weight FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```
Comment (mckent05) [inline queries.sql]: - As a result of the incorrect data you added to your database, I suggest that you rerun all transactions and queries, also please attach the correct results in your PR. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @AjroudiRida, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- You did well adding new data to your table✔️ ✔️ 
- Your PR is professional 👍 👍 
- You have done well creating and updating your animal's table 👏 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbayomiOlaoye — APPROVED** [review]
Hi @AjroudiRida  😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] To make your README more professional, kindly ensure that you remove all generic texts and comments to ensure it is carefully customized and applicable to your project. Below is a comment that should be removed from the `Author's` section:

image

Cheers, and Happy coding!👏👏👏
 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**